main.tf

```diff
@@ -261,7 +264,7 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v4
         with:
-          ref: $${{ github.event.pull_request.head.sha }}
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
Please check https://github.com/mage-os/terraform/actions/runs/12545518588/job/34979848888?pr=103. There seems to be an issue with this. I think we either need to use `${` or `$${{`.

Reply:
this escaping is very confusing to me, I'm not sure the new committed version is ok in all parts